native/src/main/java/io/ballerina/stdlib/websocket/WebSocketResourceDispatcher.java

```diff
@@ -420,12 +420,14 @@ public static void dispatchOnText(WebSocketConnectionInfo connectionInfo, WebSoc
             MethodType[] remoteFunctions = ((ServiceType) (((BValue) dispatchingService).getType())).getMethods();
             for (MethodType remoteFunc : remoteFunctions) {
                 String funcName = remoteFunc.getName();
-                if (funcName.equals(methodName) ||
-                        funcName.equals(WebSocketConstants.RESOURCE_NAME_ON_TEXT_MESSAGE) ||
-                        funcName.equals(WebSocketConstants.RESOURCE_NAME_ON_MESSAGE)) {
+                if (funcName.equals(methodName)) {
                     onTextMessageResource = remoteFunc;
                     break;
                 }
+                if (funcName.equals(WebSocketConstants.RESOURCE_NAME_ON_TEXT_MESSAGE) ||
+                        funcName.equals(WebSocketConstants.RESOURCE_NAME_ON_MESSAGE)) {
+                    onTextMessageResource = remoteFunc;
```

Comment:
we can have a `break` here right ?

Reply:
No, we can't. If there is a custom function after this, it has to be the final `onTextMessageResource`. So it has to continue in the loop.